Rate bugfix done, now the user has an array of the books that he rated a...

KarimElSheikh opened the pull request:
...nd his rating values for each book. Also, the book now stores the average and the sum of ratings and keeps them updated #82
@Vegas @Mina @Israa @omarkamal95 @Mernawaseif

KarimElSheikh committed:
Rate bugfix done, now the user has an array of the books that he rated and his rating values for each book. Also, the book now stores the average and the sum of ratings and keeps them updated

MinaNemr commented on backend/views/book.js:
```diff
@@ -286,24 +304,54 @@ module.exports = function(router) {
 			}
 
-		})
+			User.findOne({'_id': userId}, function findBookText(err, user) {
```
@KarimElSheikh 
this function name is wrong
it is not findBookText
please either remove the name or name it with another name and document it

MinaNemr commented:
@KarimElSheikh u need to fix something i mentioned on a note on book.js

Veegas commented:
@KarimElSheikh Sheel folder ./idea, w e3ml el ratings el fel user of type ObjectID

KarimElSheikh committed:
Fixed Changes Veegas and Mina mentioned in the pull request

KarimElSheikh committed:
Changed the type of the book id in the user ratings to be of type ObjectID

KarimElSheikh committed:
Revert "Changed the type of the book id in the user ratings to be of type ObjectID"

This reverts commit ba993b1448a63ebef97d63c3ed40a821b71e3973.

KarimElSheikh commented:
@MinaNemr I removed the function name, the function that call this inner function is named and documented
@Veegas I removed the .idea folder. About changing the type to ObjectID, you can always convert from the string to ObjectID using "var x = mongoose.Types.ObjectId([String])"
I didn't do it because the function Books.findOne({'_id': bookId} doesn't work if bookId is an ObjectID instead of string

Veegas commented:
@KarimElSheikh  La2 msh haynfa3 tesebha of type Mixed kda 3ashan lama hateegy te-view el books that the user rated lama teegy tegeeb el record msh hate3raf te3mlo populate

Israa-Elwan commented:
@KarimElSheikh  ur scenario are correct in your branch ! but u need to solve the conflicts in order to test the scenarios after solving them

KarimElSheikh commented:
@Veegas momken te3mel function teconvert :D
